lbrynet/core/Wallet.py

```diff
@@ -798,27 +802,35 @@ def _handle_claim_result(self, results, update_caches=True):
         defer.returnValue(results)
 
     @defer.inlineCallbacks
-    def resolve_uri(self, uri, check_cache=True):
-        cached_claim = None
-        if check_cache:
-            cached_claim = yield self._storage.get_cached_claim_for_uri(uri, check_cache)
-        if cached_claim:
-            log.debug("Using cached results for %s", uri)
-            resolve_results = cached_claim
-        else:
-            log.info("Resolving %s", uri)
-            resolve_results = yield self._get_value_for_uri(uri)
-
-        claim_id = None
-        if resolve_results and 'claim' in resolve_results:
-            claim_id = resolve_results['claim']['claim_id']
-        certificate_id = None
-        if resolve_results and 'certificate' in resolve_results:
-            certificate_id = resolve_results['certificate']['claim_id']
-
-        result = yield self._handle_claim_result(resolve_results, cached_claim is None)
-        if claim_id:
-            yield self._storage.save_claim_to_uri_cache(uri, claim_id, certificate_id)
+    def resolve(self, check_cache, page, page_size, *uris):
```

Comment:
its weird that `*uris` (the thing you care about most of the time) comes last here. it should be the first arg, and the rest should come after and have sane default values. so, something like this:

```
resolve(uris, page=1, page_size=20, check_cache=True)
```

Also, I know arrays are 0-indexed, but "page 0" makes no sense. Pagination starts on page 1.